### DISCUSSION TITLE
Backpack for Laravel starter pack

### DISCUSSION THREAD

**Discussion by christmex** [Laravel-Backpack/community-forum#501] [Show and Tell]
Hi guys, so a few days ago, I create a simple starter pack for me, and for someone who needs that 🙌✌️, this starter pack provides you a fast way to configure **backpackforlaravel** with **backpack permission**, here's the repository

https://github.com/christmex/basic-starter-pack-with-backpackforlaravel

Follow the steps and you are set, if you find any issues, let me know, thx mate

**Comment by pxpm** [comment]
Cool @christmex thanks for sharing. 

I haven't installed, just gave it a quick look, it may be even better than what I saw and it's already a cool thing that thousands of backpack projects use, so kudos to you 🙏 

I think most of the people also end up overwriting the backpack permission controllers, so it may be useful if this starter already had those controllers overwritten for easier development? I mean, create some controllers in `app/` that extend the package ones, and bind the new controllers in the service provider ? 

Did you consider that? If you did, why you decided not to do it, you don't find yourself needing that? 

It may also help to seed at least one role to add to the seeded user as an example? 

Once gain, thanks for sharing 🙏

**Comment by christmex** [comment]
Hi @pxpm , thanks for the advice 🙏

I just added:
- Default User, Role, and Permission in that starter pack.

One more thing, in my case, I don't find myself needing to overwrite the permission controllers, can you please tell me, why I should? because I don't understand why I need to do that. please let me know why 🙏

Thank you.

**Reply by pxpm** [reply]
Permissions and Roles I basically only need to add for example a description column. Sometimes I have a lot of roles and some of them can be assigned through the admin panel while others don't, so I create scopes and accessors there.

Most diverse changes are in User controller, add fields, filters etc. 

This was asked today on SO https://stackoverflow.com/questions/76330453/laravel-backpack-issue-adding-route-for-crud-controller

And i've seen this question done quite some times. 

There is no way for us (backpack) to do what you can do when developing a starter skeleton or similar because you are creating files in "user land".

Once again, thanks 🙏

**Reply by christmex** [reply]
Hi @pxpm thank you so much for your explanation 🙏, I've already updated the starter pack with add a few things, such as

- extending the controller for user, role, and permission (so then can be more flexible)
- extending the model for user, role, and permission (so then can be more flexible)
- add Post CRUD as an example 

I also tried to apply the **Policy Concept** but I ended up facing a problem, 
```php
    protected function setupListOperation()
    {
        $this->authorize('view', backpack_auth()); //cause error, authorize doesn't exist
        CRUD::addColumn([
            'name' => 'user_id',
            'type' => 'select',
            'entity'    => 'user', // the method that defines the relationship in your Model
            'attribute' => 'name', // foreign key attribute that is shown to user
            'model'     => "App\Models\User", // foreign key model
        ]);
        CRUD::column('title');
        CRUD::column('description');

        /**
         * Columns can be defined using the fluent syntax or array syntax:
         * - CRUD::column('price')->type('number');
         * - CRUD::addColumn(['name' => 'price', 'type' => 'number']); 
         */
    }
```

if I use this  to check I got  **authorize doesn't exist** error
```php
$this->authorize('view', backpack_auth())
```
<img width="914" alt="image" src="https://github.com/Laravel-Backpack/community-forum/assets/34405837/9c312e36-fdf4-4e40-9fa6-7fefccf41720">

or should I add this?
```php
use Illuminate\Foundation\Auth\Access\AuthorizesRequests as authorize;
```

Or did I miss something, did  I make a mistake with my Policy Concept? because I'm still trying to understand this concept, please help me out.

Again thank you so much for helping me and making me understand it more 🙏
